src/FluentAssertions.Web/Internal/JsonExtensions.cs

```diff
@@ -1,68 +1,94 @@
-﻿using Newtonsoft.Json.Linq;
-using System;
+﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Text.Json;
 
 namespace FluentAssertions.Web.Internal
 {
     internal static class JsonExtensions
     {
-        public static bool HasKey(this JToken json, string key)
+        public static bool HasKey(this JsonDocument json, string key)
             => json.GetByKey(key).Any();
 
-        public static IEnumerable<string> GetStringValuesByKey(this JToken json, string key)
+        public static IEnumerable<string> GetStringValuesByKey(this JsonDocument json, string key)
         {
             var byKey = json.GetByKey(key);
-            if (byKey != null && !byKey.Any())
+
+            if (!byKey.Any())
             {
                 return Enumerable.Empty<string>();
             }
 
-            var first = byKey.FirstOrDefault()?.First;
-            return first switch
+            var first = byKey.First();
+            switch (first.Value.ValueKind)
             {
-                JArray values => values.Select(c => (string)c),
-                { } token => new[] { token.ToString() },
-                _ => Enumerable.Empty<string>()
-            };
+                case JsonValueKind.Array:
+                    return first.Value.EnumerateArray().Select(c => c.ToString() ?? "");
+                default:
```

Comment:
Add tests to check the behavior for properties with complex values and with null, undefined values.